ruby/ql/lib/codeql/ruby/security/internal/CleartextSources.qll

```diff
@@ -153,18 +159,19 @@ module CleartextSources {
   }
 
   /**
-   * A value written to a hash entry with a key that may contain password information.
+   * A value written to a hash entry with a key that may contain sensitive information.
    */
-  private class HashKeyWritePasswordSource extends Source {
+  private class HashKeyWriteSensitiveSource extends Source {
     private string name;
     private DataFlow::ExprNode recv;
 
-    HashKeyWritePasswordSource() {
-      exists(DataFlow::CallNode writeNode |
-        name.regexpMatch(maybePassword()) and
+    HashKeyWriteSensitiveSource() {
+      exists(DataFlow::CallNode writeNode, SensitiveDataClassification classification |
+        nameIndicatesSensitiveData(name, classification) and
+        not classification = SensitiveDataClassification::id() and
```

Comment:
Nitpick: I'd prefer an inclusion test to make it more obvious which categories are considered, and maybe make this a less likely to change inadvertently if we add more classifications in the future, e.g.

```suggestion
        classification = [
          SensitiveDataClassification::secret(),
          SensitiveDataClassification::password(),
          SensitiveDataClassification::certificate(),
          SensitiveDataClassification::private()
        ] and
```